nextcloudappstore/core/api/v1/views.py

```diff
@@ -130,3 +133,30 @@ def get_object(self):
         release = get_object_or_404(release)
         self.check_object_permissions(self.request, release)
         return release
+
+
+class SessionObtainAuthToken(APIView):
+    """Modified version of rest_framework.authtoken.views.ObtainAuthToken.
+
+    Modified to return token based on SessionAuthentication, and not just data
+    sent for BasicAuthentication.
+    """
+
+    throttle_classes = (PostThrottle,)
+    permission_classes = (IsAuthenticated,)
+    parser_classes = (parsers.FormParser,
+                      parsers.MultiPartParser,
+                      parsers.JSONParser,)
+    renderer_classes = (renderers.JSONRenderer,)
+    serializer_class = AuthTokenSerializer
+
+    def post(self, request, *args, **kwargs):
+        if request.user.is_authenticated():
```

Comment:
I assume it would not return true if we couldn't? From REST API docs:
> If successfully authenticated, SessionAuthentication provides the following credentials.
>
> - request.user will be a Django User instance.
> - request.auth will be None.

http://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication

I think `post()` is one of the last methods called in the class-
based view.